### ISSUE TITLE
Images with spaces in the filename are not served

### ISSUE THREAD

**Issue by fra967** [dadi/cdn#57]
Spaces are encoded as %20 by the browser, but images are not found on the filesystem (where they are saved unencoded)

Example:
this image with spaces in the filename is not being shown
http://bantam.test.empireonline.com:8080/jpg/70/0/0/640/480/aspectfit/0/0/0/0/0/0/c/articles/578f6fa76c7bafe3054f1635/Finding%20Dory%20-%20Seals.jpg

but it exists on the filesystem
```# ls -l /data/asset-store/c/articles/578f6fa76c7bafe3054f1635/Finding\ Dory\ -\ Seals.jpg 
-rw-r--r-- 1 www-data www-data 150728 Jul 20 15:20 /data/asset-store/c/articles/578f6fa76c7bafe3054f1635/Finding Dory - Seals.jpg

Please note that previous versions of CDN were able to serve images, for example this is 0.1.11-beta
http://54.229.5.231:8080/jpg/70/0/0/640/480/0/0/0/1/0/North/0/0/0/0/0/c/celebrity/5698144c01aa022c6edd2986/DRAKE%20WHO%20WORE%20IT%20BEST_950x953.png

**Comment by dark12222000** [comment]
Looks like the Disk adapter is just missing a few calls to urlencode()

**Comment by fra967** [comment]
I am re-opening as it seems a related encoding issue, this time with a file containing parentheses in the file name
http://bantam.test.empireonline.com:8080/jpg/70/0/0/640/480/aspectfit/0/0/0/0/0/0/c/reviews_films/57619712823490d062647d2e/Rotating_earth_%28large%29.gif
```
$ ls -l /data/asset-store/c/reviews_films/57619712823490d062647d2e/Rotating_eart*
-rw-r--r-- 1 www-data www-data 1429302 Jul 20 17:33 /data/asset-store/c/reviews_films/57619712823490d062647d2e/Rotating_earth_(large).gif
```

**Comment by fra967** [comment]
Closing as it is a different issue (animated gif support)

**Comment by eduardoboucas** [comment]
Just a quick note to say that the image processing library we currently use doesn't support animated gifs, and there's an [open issue](https://github.com/EyalAr/lwip/issues/153) for it. If we need this, we'll need to get our hands dirty! 💪

**Comment by jimlambie** [comment]
The issue here was due to stream backpressure when trying to get EXIF data from images. It is resolved in v1.4.7.

We support animated GIFs insofar as we will return the first frame only.
